nautobot/ipam/api/views.py

```diff
@@ -405,36 +411,86 @@ def available_vlans(self, request, pk=None):
         By default, the number of VIDs returned will be equivalent to PAGINATE_COUNT.
         An arbitrary limit (up to MAX_PAGE_SIZE, if set) may be passed, however results will not be paginated.
         """
-        obj = get_object_or_404(self.queryset, pk=pk)
+        vlan_group = get_object_or_404(self.queryset, pk=pk)
 
-        try:
-            limit = int(request.query_params.get("limit", get_settings_or_config("PAGINATE_COUNT")))
-        except ValueError:
-            limit = get_settings_or_config("PAGINATE_COUNT")
+        if request.method == "POST":
+            with cache.lock(
+                "nautobot.ipam.api.views.available_vlans", blocking_timeout=5, timeout=settings.REDIS_LOCK_TIMEOUT
+            ):
+                # Normalize to a list of objects
+                requested_vlans = request.data if isinstance(request.data, list) else [request.data]
+
+                # Determine if the requested number of VLANs is available
+                available_vids = vlan_group.available_vids
+                if len(available_vids) < len(requested_vlans):
+                    return Response(
+                        {
+                            "detail": (
+                                f"An insufficient number of VLANs are available within the VLANGroup {vlan_group} "
+                                f"({len(requested_vlans)} requested, {len(available_vids)} available)"
+                            )
+                        },
+                        status=status.HTTP_204_NO_CONTENT,
+                    )
+
+                # Assign VLAN IDs from the list of VLANGroup's available VLAN IDs.
+                _available_vids = iter(available_vids)
+
+                for requested_vlan in requested_vlans:
+                    if "vid" not in requested_vlan:
+                        requested_vlan["vid"] = f"{next(_available_vids)}"
+
+                    requested_vlan["vlan_group"] = vlan_group.pk
+
+                # Initialize the serializer with a list or a single object depending on what was requested
+                context = {"request": request, "depth": 0}
+                serializer = serializers.VLANSerializer(data=requested_vlans, many=True, context=context)
```

Comment:
In `available-prefixes`/`available-ips` we only use `many=True` for the response if the user requested more than one allocation. Is that desirable here as well, or is keeping the API response more predictably structured (always a list/many response) to be preferred, do you think?

Reply:
Today endpoint accepts a single dict or list of dicts, but in each cases responds with a list.
I think this is more predictable, especially in context of count / next / previous / results keys we added to the Response

Reply:
OK, now I think it makes it more consistent to respond in the request format and similar to existing endpoints. 👍